### ISSUE TITLE
THREE.ArrowHelper use accross several contexts fails

### ISSUE THREAD

**Issue by ebann** [mrdoob/three.js#5871]
It seems that the last changes made on this class introduced a bug/limitation.

I can't have 2 different canvas with some arrows created by this class anymore.
Here is the warning raised by webgl:
 WebGL: INVALID_OPERATION: bindBuffer: object not from this context

My guess it that the optimization lead to sharing the buffers across different GLContexts.

It also leads to incorrect rendering of the arrows.

**Comment by mrdoob** [comment]
Uh oh! :(

**Comment by ebann** [comment]
Using the previous implementation fix the issue though

**Comment by vicapow** [comment]
I just ran into this myself, as well. This is caused by the geometry being shared across contexts outside of the closure in ArrowHelper's definition.

````
THREE.ArrowHelper = ( function () {
        /* create geometry once. */
	return function ( dir, origin, length, color, headLength, headWidth ) {
            /* reuse geometry */
        };
})();
````

This can be fixed by recreating the geometry for each arrow but would probably be a performance regression for most use-cases:

````
THREE.ArrowHelper = ( function () {
	return function ( dir, origin, length, color, headLength, headWidth ) {
            /* create geometry. */
            /* use geometry. */
        };
})();
````

One possibility is to pass an additional argument, something like: "shouldShareGeometry" but this feels dirty. I'm happy to try and submit a diff on for whatever is decided on.

**Comment by mrdoob** [comment]
See #6062.

**Comment by ChrisFichman** [comment]
Seems like this is still an issue in r71. 

It happens to me with the arrow helper, as per @ebann 's description, when I'm changing contexts in a node.js app with a backbone framework, but I suspect it's the same issue. 

I used vicapow's fix and it worked. My performance didn't take a hit, but I'm only drawing a few ArrowHelpers. I can imagine someone trying to draw something like a vector field would have a problem though.

**Comment by mrdoob** [comment]
I think this should be now fixed in the `dev` branch. Could someone double check?

**Comment by dubejf** [comment]
Fixed by #6723.

 ![5871](https://cloud.githubusercontent.com/assets/8649151/8512566/b505c5d2-2316-11e5-802f-46a093db290f.png)
